src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1952,15 +1939,17 @@ MulticopterPositionControl::control_position(float dt)
 
 			} else {
 				/* Z component is negative, going down, simply limit thrust vector */
-				float k = thr_max / thrust_abs;
+				float k = thr_max / thrust_body_z;
 				thrust_sp *= k;
```

Comment:
That's an error, thanks for spotting.
I just saw that there's something fishy with https://github.com/PX4/Firmware/pull/6205/files#diff-fc77c3ef569029d45764664c75d4b0c1R1793.